src/viewer.js

```diff
@@ -1575,11 +1575,15 @@ $.extend( $.Viewer.prototype, $.EventSource.prototype, $.ControlDock.prototype,
                 _this._loadQueue.splice(0, 1);
 
                 if (queueItem.options.replace) {
-                    var newIndex = _this.world.getIndexOfItem(queueItem.options.replaceItem);
+                    const replaced = queueItem.options.replaceItem;
+                    const newIndex = _this.world.getIndexOfItem(replaced);
                     if (newIndex !== -1) {
                         queueItem.options.index = newIndex;
                     }
-                    _this.world.removeItem(queueItem.options.replaceItem);
+                    if (!replaced._zombieCache && replaced.source.equals(queueItem.tileSource)) {
```

Comment:
This is nice feature that we allow implicit usage of zombies in case the sources equal & tile is being replaced. Zombies used automatically when suitable. The problem is `equals` implementation accross all sources. For now I just check base URLs but query params might vary a lot. Maybe try generating tile URL and copare these URLS? But how do we know particular tile exists? Might not know even the `equals` function itself..